lib/wanda_web/views/v1/execution_json.ex

```diff
@@ -1,26 +1,25 @@
-defmodule WandaWeb.V1.ExecutionView do
-  use WandaWeb, :view
+defmodule WandaWeb.V1.ExecutionJSON do
+  alias Wanda.Executions.Execution
+  alias WandaWeb.V1.ExecutionJSON
 
-  alias WandaWeb.V1.ExecutionView
-
-  def render("index.json", %{executions: executions, total_count: total_count}) do
+  def index(%{executions: executions, total_count: total_count}) do
     %{
-      items: render_many(executions, ExecutionView, "execution.json"),
+      items: Enum.map(executions, &execution/1),
       total_count: total_count
     }
   end
 
-  def render("show.json", %{execution: execution}) do
-    render_one(execution, ExecutionView, "execution.json")
-  end
-
-  def render("execution.json", %{execution: execution}) do
+  def execution(%Execution{} = execution) do
     execution
-    |> render_one(WandaWeb.V2.ExecutionView, "execution.json")
+    |> WandaWeb.V2.ExecutionJSON.execution()
     |> adapt_v1()
   end
 
-  def render("start.json", %{
+  def show(%{execution: execution}) do
```

Comment:
I would put this `show` after `index` (actually, you don't need the `ExecutionJSON` prefix here), and make the `def execute` a private function, as it is only used here. I guess you cannot make them private `v2` on, as they are used here